.github/workflows/cppcheck.yaml

```diff
@@ -0,0 +1,25 @@
+name: Static analysis with cppcheck
```

Comment:
Let's rename it to check_code, or static_analysis to do not rename this file if we change the tool in the future.